components/elm/src/biogeochem/EcosystemDynMod.F90

```diff
@@ -144,115 +144,106 @@ subroutine EcosystemDynLeaching(bounds, num_soilc, filter_soilc, &
     integer                  , intent(in)    :: filter_pcropp(:)  ! filter for prognostic crop patches
     logical                  , intent(in)    :: doalb             ! true = surface albedo calculation time step
     type(cnstate_type)       , intent(inout) :: cnstate_vars
-    type(carbonflux_type)    , intent(inout) :: carbonflux_vars
-    type(carbonstate_type)   , intent(inout) :: carbonstate_vars
-    type(carbonflux_type)    , intent(inout) :: c13_carbonflux_vars
-    type(carbonstate_type)   , intent(inout) :: c13_carbonstate_vars
-    type(carbonflux_type)    , intent(inout) :: c14_carbonflux_vars
-    type(carbonstate_type)   , intent(inout) :: c14_carbonstate_vars
-    type(nitrogenflux_type)  , intent(inout) :: nitrogenflux_vars
-    type(nitrogenstate_type) , intent(inout) :: nitrogenstate_vars
-    type(waterstate_type)    , intent(in)    :: waterstate_vars
-    type(waterflux_type)     , intent(in)    :: waterflux_vars
     type(frictionvel_type)   , intent(in)    :: frictionvel_vars
     type(canopystate_type)   , intent(inout) :: canopystate_vars
 
-    type(phosphorusflux_type)  , intent(inout) :: phosphorusflux_vars
-    type(phosphorusstate_type) , intent(inout) :: phosphorusstate_vars
-
+    character(len=64) :: event
+    real(r8) :: dt
     !-----------------------------------------------------------------------
-  
-    !if(.not.(use_pflotran.and.pf_cmode)) then
-    call t_startf('PhosphorusWeathering')
+    dt = dtime_mod;
+
+    ! only do if ed is off
+    event = 'PhosphorusWeathering'
+    call t_startGPU(event)
     call PhosphorusWeathering(num_soilc, filter_soilc, &
-         cnstate_vars,phosphorusstate_vars,phosphorusflux_vars)
-    call t_stopf('PhosphorusWeathering')
-    
-    call t_startf('PhosphorusAdsportion')
+         cnstate_vars, dt)
+    call t_stopGPU(event)
```

Comment:
`t_*GPU` is a misleading name as the subroutine is applicable for CPU and GPU. An alterante may be `t_*_lnd`. @sarats may have better suggestions.

Reply:
Not sure what t_stopGPU does. Could you clarify? Are you doing GPU profiling through CUPTI? Is there a missing start('PhosphorusAdsportion') or just moved elsewhere?

Reply:
The idea for these timers was to be a simple wrapper to be used by anyone doing GPU porting to leave the CPU timers unchanged.  I thought it might be better to eventually include a routine like this in perfMod itself for any future E3SM gpu development work, which is why I used the generic *GPU name.  But, I'm happy renaming it to whatever fits best.

For profiling, I have been using nvprof and then switched to Nsight.  I do think that adding API calls or adding custom device timing routines to this timer is a great idea.  The profiling in E3SM as a whole is very messy especially and given module/routine level information is unreliable since nvprof has been deprecated.  Unit testing is helpful but it doesn't provide an accurate picture of which routines are bottlenecks.